src/main/java/com/structures/helpers/Structure.java

```diff
@@ -0,0 +1,422 @@
+        }
+
+        for (final Entity anEntityList : entityList)
+        {
+            Minecraft.getMinecraft().getRenderManager().renderEntityStatic(anEntityList, 0.0F, true);
+        }
+    }
+
+    private void renderGhost(final World world, ModelHolder holder, final EntityPlayer player, final float partialTicks)
+    {
+        boolean existingModel = !this.mc.theWorld.isAirBlock(holder.pos);
+
+        IBlockState actualState = holder.actualState;
+        Block block = actualState.getBlock();
+
+        if (actualState.getRenderType() == EnumBlockRenderType.MODEL)
+        {
+            BlockRenderLayer originalLayer = MinecraftForgeClient.getRenderLayer();
+
+            for (BlockRenderLayer layer : BlockRenderLayer.values())
+            {
+                if (block.canRenderInLayer(actualState, layer))
+                {
+                    this.mc.getTextureManager().bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+                    ForgeHooksClient.setRenderLayer(layer);
+                    this.renderGhostBlock(world, holder, player, layer, existingModel, partialTicks);
+                    holder.rendered = true;
+                }
+            }
+
+            ForgeHooksClient.setRenderLayer(originalLayer);
+        }
+
+        if (holder.te != null && !holder.rendered)
+        {
+            TileEntity te = holder.te;
+            te.setPos(holder.pos);
+            FakeWorld fakeWorld = new FakeWorld(holder.actualState, world.getSaveHandler(), world.getWorldInfo(), world.provider, world.theProfiler, true);
+            te.setWorldObj(fakeWorld);
+            int pass = 0;
+
+            if (te.shouldRenderInPass(pass))
+            {
+                TileEntityRendererDispatcher terd = TileEntityRendererDispatcher.instance;
+                terd.func_190056_a(fakeWorld,
+                        Minecraft.getMinecraft().renderEngine,
+                        Minecraft.getMinecraft().fontRendererObj,
+                        new FakeEntity(fakeWorld),
+                        (RayTraceResult) null,
+                        0.0F);
+                GL11.glPushMatrix();
+                terd.renderEngine = Minecraft.getMinecraft().renderEngine;
+                terd.preDrawBatch();
+                GL11.glColor4f((float) (this.COLOR_R / COLOR_R), (float) (this.COLOR_G / COLOR_G), (float) (this.COLOR_B / COLOR_B), (this.COLOR_A / COLOR_A));
+                terd.renderTileEntity(te, partialTicks, -1);
+                terd.drawBatch(pass);
+                GL11.glPopMatrix();
+            }
+        }
+    }
+
+    private void renderGhostBlock(final World world, ModelHolder holder, final EntityPlayer player, BlockRenderLayer layer, boolean existingModel, final float partialTicks)
+    {
+        double dx = player.lastTickPosX + (player.posX - player.lastTickPosX) * partialTicks;
+        double dy = player.lastTickPosY + (player.posY - player.lastTickPosY) * partialTicks;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'dy' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)